pkg/policy/utils.go

```diff
@@ -11,3 +14,53 @@ func isRunningPod(obj unstructured.Unstructured) bool {
 
 	return phase == "Running"
 }
+
+// check if all slice elements are same
+func isMatchResourcesAllValid(slice []string) bool {
+	if len(slice) == 0 {
+		return false
+	}
+	for i := 1; i < len(slice); i++ {
+		if slice[i] != slice[0] {
+			return false
+		}
+	}
+	return true
+}
+
+func fetchUniqueKinds(rule kyverno.Rule) []string {
+	var kindlist []string
+
+	kindlist = append(kindlist, rule.MatchResources.Kinds...)
+
+	for _, all := range rule.MatchResources.Any {
+		kindlist = append(kindlist, all.Kinds...)
+	}
+
+	for _, all := range rule.MatchResources.All {
```

Comment:
Don't you need to check `isMatchResourcesAllValid` before adding `rule.MatchResources.All.kinds` to the list?

Reply:
Yes we can , done